app/ui/src/app/integration/edit-page/action-configure/action-configure.component.ts

```diff
@@ -196,8 +197,9 @@ export class IntegrationConfigureActionComponent implements OnInit, OnDestroy {
   initForm(position: number, page: number, descriptor: ActionDescriptor, error?: any) {
     if (error) {
       this.error = error;
-      this.error.message = error.message || error.userMsg || error.developerMsg;
       this.error.class = 'alert alert-warning';
+      this.error.icon = 'pficon pficon-warning-triangle-o';
+      this.error.message = error.message || error.userMsg || error.developerMsg;
```

Comment:
it's usually not a good idea to mix presentation logic with interaction/data/business logic inside the component controller. This makes editing the UI harder. In this particular case, we're forcing the template to feature certain classnames depending on the value of `error`. It would be easier to define classnames like this then:

```
<p [ngClass]="{ 'alert alert-warning': error }">
  <span [ngClass]="{ 'pficon pficon-warning-triangle-o': error }"></span>
  {{ error?.message || error?.userMsg || error?.developerMsg }}
</p>
```

That would allow to remove all the LOCs above just leave `this.error = error;` inside the conditional

Reply:
@deeleman it looks like there are 2 types of errors for this. An "alert" and a "warning". If it's an "alert", we use an alert class on the output and display an alert icon and a string of text. If it's a "warning", we use a warning class on the output and display a warning icon and either `error?.message || error?.userMsg || error?.developerMsg`. How would you suggest doing it with those variations?

Reply:
We need to elaborate this a bit. How do we tell wether it is an alert or a warning? Please point me to the object variable that signals what case scenario is and I'll do my best to propose a way forward. Also, I'd need to know what DOM elements should we render for each case.

Reply:
It's one of many areas in the editor that needs work, effectively it's set here if there's an error or if there's no properties -> https://github.com/syndesisio/syndesis/blob/master/app/ui/src/app/integration/edit-page/action-configure/action-configure.component.ts#L200-L217